docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,110 @@
+3. And please file a bug report to help us figure out how to generate better SQL.
+
+### Values are repeatedly being converted on the fly.
+
+Low performance when using Metabase can also be caused by incorrect typing of columns, e.g., by storing a numeric value as a string.  When this happens, the query converts values on the fly each time the query is run.
+
+**How to detect this:** Even a handwritten query will be slow if it has to do this conversion every time. You can also spot this by looking at the raw data types of the columns being used in the query.
+
+**How to fix this:** Amend the database schema to store numbers as numbers, timestamps as timestamps, and so on, rather than as strings or other data types.
+
+### The Java Virtual Machine is using too much memory.
+
+We discuss this problem and its solutions in [this troubleshooting guide](./running.md).
+
+### Metabase is running on an under-powered machine.
+
+**How to detect this:** Checking the performance logs for the server where Metabase is running will tell you whether it is hitting CPU or memory limits. However, it's much more likely that the database itself is hitting its limits, so please check it first.
+
+**How to fix this:** Upgrade to a more powerful server, or one with more memory.
+
+Note: If you are running in the cloud, the servers we select are enough for 98% of customers: we use a 2-node cluster for all our in-house analytics and a 3-node cluster for customers that autoscales whenever performance requires it. If you are using our cloud option and the Metabase server really is the problem, you should consider deploying your own instance.
+
+### A dashboard contains too many questions.
+
+When Metabase displays a dashboard, it re-runs all of the questions. We do our best to do this concurrently, and the network layers and the database itself also do what they can, but a dashboard with a hundred cards is going to be slower than a single question. And if your dashboard contains filters, then each time someone changes a filter setting, all of the cards that depend on it have to re-execute. Careful dashboard design can prevent or eliminate these problems.
+
+**How to detect this:** The individual questions load quickly when viewed on their own, but the dashboard loads or updates slowly.
+
+**How to fix this:** See [this article][faster-dashboards] for tips on making dashboards more performant.
+
+### The UI appears to freeze when saving a question that has not yet been run
+
+**How to detect this:** If you save a question that has not been executed, MB runs the question while saving, which can make the UI look frozen.
+
+**How to fix this:** This is [a bug](https://github.com/metabase/metabase/issues/14957) and we are working to fix it. Until it's corrected, the workaround is to run the question before saving it.
```

Comment:
There's a greater chance that people are hitting another problem than that specific bug. More likely lack of appdb connections, full Jetty queue, small network hiccups, etc.

Reply:
incorporated